python/cudf/cudf/tests/test_string.py

```diff
@@ -386,16 +417,26 @@ def test_string_cat(ps_gs, others, sep, na_rep, index):
     ps, gs = ps_gs
 
     pd_others = others
-    if isinstance(pd_others, pd.Series):
-        pd_others = pd_others.values
-    if isinstance(pd_others, (list, tuple)):
-        for elem in pd_others:
-            if isinstance(elem, (pd.Series, Series)):
-                elem.index = ps.index
+    if isinstance(pd_others, (pd.Series, pd.Index)):
+        gd_others = cudf.from_pandas(pd_others)
+    else:
+        gd_others = pd_others
 
-    expect = ps.str.cat(others=pd_others, sep=sep, na_rep=na_rep)
-    got = gs.str.cat(others=others, sep=sep, na_rep=na_rep)
+    if isinstance(gd_others, (list, tuple)):
+        temp_tuple = []
+        for elem in gd_others:
+            if isinstance(elem, (pd.Series, pd.Index)):
+                temp_tuple.append(cudf.from_pandas(elem))
+            else:
+                temp_tuple.append(elem)
+
+        if isinstance(gd_others, tuple):
+            gd_others = tuple(temp_tuple)
+        else:
+            gd_others = list(temp_tuple)
```

Comment:
Can you explain what this is doing?

Reply:
This part of code is converting any `pd` objects(`pd.Index`, `pd.Series`) to `cudf` objects(`cudf.Index`, `cudf.Series`). Simplified it a bit now